lib/compiler/passes/generate-bytecode.js

```diff
@@ -590,48 +615,30 @@ function generateBytecode(ast) {
           [op.FAIL, expectedIndex]
         );
       } else {
-        const stringIndex = addConst("\"\"");
+        const stringIndex = addLiteralConst("");
 
         return [op.PUSH, stringIndex];
       }
     },
 
     class(node) {
-      const regexp = "/^["
-        + (node.inverted ? "^" : "")
-        + node.parts.map(part =>
-            Array.isArray(part)
-              ? js.regexpClassEscape(part[0])
-                + "-"
-                + js.regexpClassEscape(part[1])
-              : js.regexpClassEscape(part)
-          ).join("")
-        + "]/" + (node.ignoreCase ? "i" : "");
-      const parts = "["
-        + node.parts.map(part =>
-            Array.isArray(part)
-              ? "[\"" + js.stringEscape(part[0]) + "\", \"" + js.stringEscape(part[1]) + "\"]"
-              : "\"" + js.stringEscape(part) + "\""
-          ).join(", ")
-        + "]";
-      const regexpIndex = addConst(regexp);
-      const expectedIndex = addConst(
-        "peg$classExpectation("
-          + parts + ", "
-          + node.inverted + ", "
-          + node.ignoreCase
-          + ")"
-      );
+      const classIndex = addClassConst(node);
```

Comment:
OK, this is the heart of the change.  I'm not quite sure why all of the consts needed to be split to get to this.  However, I really like *this* change, so I'm willing to take all the rest.

Reply:
Actually, the _heart_ is holding functions in the different array. _That_ was the key of all changes, because it allows you to generate code for other languages without parsing JS function definitions to insert types, for example.